File: backend/gamelink/src/main/java/urjc/gamelink/Service/NewsService.java
package urjc.gamelink.Service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.boot.autoconfigure.data.web.SpringDataWebProperties.Pageable;
//import org.springframework.data.domain.Page;
//import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import urjc.gamelink.Model.News;
import urjc.gamelink.Repositories.NewRepository;

@Service
public class NewsService {
    
	@Autowired
	private NewRepository repository;

	public Optional<News> findById(long id) {
		return repository.findById(id);
	}
	
	public boolean exist(long id) {
		return repository.existsById(id);
	}

	public List<News> findAll() {
		return repository.findAll();
	}

	public void save(News n) {
		repository.save(n);
	}

	public void delete(long id) {
		repository.deleteById(id);
	}

    public List<News> findByIds(List<Long> notices) {
        return repository.findAllById(notices);
    }

}

File: frontend/src/app/Service/NewsService.ts
import { Service } from '@angular/core';
import { HttpClient } from '@angular/common/http';

import { News } from '../Model/News';
import { NewRepository } from '../Repositories/NewRepository';

@Service
export class NewsService {

    constructor(private httpClient: HttpClient) {}

    private repository: NewRepository;

    findById(id: number) {
        return this.repository.findById(id);
    }

    exist(id: number) {
        return this.repository.existById(id);
    }

    findAll(pageable: Pageable) {
        return this.repository.findAll(pageable);
    }

    findAll() {
        return this.repository.findAll();
    }

    save(new: News) {
        return this.httpClient.put(URL + new.id, new);
    }

    delete(id: number) {
        this.repository.deleteById(id);
    }

    findAll(of: PageRequest) {
        return this.repository.findAll(of);
    }

    findByIds(notices) {
        return this.repository.findAllById(notices);
    }
}